### DISCUSSION TITLE
Lags with animation

### DISCUSSION THREAD

**Discussion by Thoronion** [Mapsui/Mapsui#2666] [Q&A]
Hi.

I have a small lag when using a layer other than RasterizingTileLayer along with animations. See the GIF below. The feeling is that the map is moved first and the features afterwards with a noticable delay. At first I thought this was related to the size of the features but even with small amount of features the lag persists. With RasterizingTileLayer the lag is gone. RasterizingLayer has the lag as well. Could this be a bug with the renderer? I don't think it is performance related.

![test](https://github.com/Mapsui/Mapsui/assets/18140541/e6065ad7-4b80-492d-b516-06778eb97123)

**Comment by charlenni** [comment]
I would say, that the LineString you use need some time to draw.

Did you tried to use the performance widget to see, how much time the last drawing needs?

**Reply by Thoronion** [reply]
It takes 27 ms to draw the LineString above according to the widget. I tried with a much smaller LineString and it took 5 ms to draw and it had the same lagging effect.

**Comment by pauldendulk** [comment]
Every render cycle we draw everything from scratch, so the speed with which it is drawn should not be an issue.

One thing that could cause this is pixel alignment of the tile layer. In the renderer we have some logic to make sure that pixels in an image are drawn 1 on 1 to pixels on screen. Not sure how important this still is now that pixel resolutions are so high. We could turn it of to see if this fixes the problem. If pixel alignment looks significantly better we could apply the same alignment to all layers, even if they don't need it. There should be a single base layer that is leading.

Another explanation is that there is some float to integer rounding somewhere.
